### PR TITLE
chore: added coverage for ts files

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change
- This pull request introduces code coverage for TypeScript files in the repository. As part of the ongoing migration from JavaScript to TypeScript, it is important to ensure that the TypeScript code is adequately covered by tests

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: jest.config.typescript.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ module.exports = {
     // collectCoverageFrom: null,
 
     // The directory where Jest should output its coverage files
-    coverageDirectory: 'reports/coverage',
+    coverageDirectory: 'reports/ts-coverage',
```
Comment (saikumarrs) [inline jest.config.typescript.js:27]: You'd have to add this coverage directory to the sonar properties file. Also, we need to update the GitHub workflow (report-code-coverage.yml), where we update the filesystem paths.

Additionally, we need to upload (report-code-coverage.yml) the new coverage reports to Codecov.

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: jest.config.typescript.js (hunk 1 above)
Comment (mihir-4116) [reply jest.config.typescript.js:27]: updated

File: .github/workflows/report-code-coverage.yml [hunk 2]
```diff
@@ -41,11 +42,17 @@ jobs:
         uses: codecov/codecov-action@v3.1.4
         with:
           directory: ./reports/coverage
+      
+      - name: Upload Coverage Reports to Codecov
```
Comment (saikumarrs) [inline .github/workflows/report-code-coverage.yml:46]:
```suggestion
      - name: Upload TS Coverage Reports to Codecov
```

Comment (saikumarrs) [reply .github/workflows/report-code-coverage.yml:46]: Also, did this work? I mean, did Codecov aggregate all the coverage data?

**Review by saikumarrs — CHANGES_REQUESTED** [review]

**Review by utsabc — APPROVED** [review]

File: .github/workflows/report-code-coverage.yml (hunk 2 above)
Comment (mihir-4116) [reply .github/workflows/report-code-coverage.yml]: Step title updated and yes, Codecov aggregated all the coverage data

**Review by saikumarrs — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**